google/cloud/spanner_v1/testing/database_test.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2023 Google LLC All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import grpc
+
+from google.api_core import grpc_helpers
+import google.auth.credentials
+from google.cloud.spanner_admin_database_v1 import DatabaseDialect
+from google.cloud.spanner_v1 import SpannerClient
+from google.cloud.spanner_v1.database import Database, SPANNER_DATA_SCOPE
+from google.cloud.spanner_v1.services.spanner.transports import (
+    SpannerGrpcTransport,
+    SpannerTransport,
+)
+from google.cloud.spanner_v1.testing.interceptors import (
+    MethodCountInterceptor,
+    MethodAbortInterceptor,
+)
+
+
+class TestDatabase(Database):
+    """Representation of a Cloud Spanner Database. This class is only used for
+    system testing as there is no support for interceptors in grpc client
+    currently, and we don't want to make changes in the Database class for
+    testing purpose as this is a hack to use interceptors in tests."""
+
+    def __init__(
+        self,
+        database_id,
+        instance,
+        ddl_statements=(),
+        pool=None,
+        logger=None,
+        encryption_config=None,
+        database_dialect=DatabaseDialect.DATABASE_DIALECT_UNSPECIFIED,
+        database_role=None,
+        enable_drop_protection=False,
+    ):
+        super().__init__(
+            database_id,
+            instance,
+            ddl_statements,
+            pool,
+            logger,
+            encryption_config,
+            database_dialect,
+            database_role,
+            enable_drop_protection,
+        )
+
+        self._method_count_interceptor = MethodCountInterceptor()
+        self._method_abort_interceptor = MethodAbortInterceptor()
+        self._interceptors = [
+            self._method_count_interceptor,
+            self._method_abort_interceptor,
+        ]
+
+    @property
+    def spanner_api(self):
+        """Helper for session-related API calls."""
```

Comment:
nit: Maybe update this comment, as this client is not necessarily only for session-related API calls. It could for example also be used to manually execute a `Rollback` request.

Reply:
As discussed not needed